torchgeo/datasets/deepglobelandcover.py

```diff
@@ -0,0 +1,267 @@
+    and valid dataset are not accompanied by labels. The dataset that we use with a
+    custom train/test split can be downloaded from `Kaggle <https://www.kaggle.com/
+    datasets/geoap96/deepglobe2018-landcover-segmentation-traindataset>`__ (created as a
+    part of Computer Vision by Deep Learning (CS4245) course offered at TU Delft).
+
+    Dataset format:
+
+    * images are RGB data
+    * masks are RGB image with with unique RGB values representing the class
+
+    Dataset classes:
+
+    0. Urban land
+    1. Agriculture land
+    2. Rangeland
+    3. Forest land
+    4. Water
+    5. Barren land
+    6. Unknown
+
+    File names for satellite images and the corresponding mask image are id_sat.jpg and
+    id_mask.png, where id is an integer assigned to every image.
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/pdf/1805.06561.pdf
+
+    """
+
+    filename = "data.zip"
+    data_root = "data"
+    md5 = "f32684b0b2bf6f8d604cd359a399c061"
+    splits = ["train", "test"]
+    classes = [
+        "Urban land",
+        "Agriculture land",
+        "Rangeland",
+        "Forest land",
+        "Water",
+        "Barren land",
+        "Unknown",
+    ]
+    colormap = [
+        (0, 255, 255),
+        (255, 255, 0),
+        (255, 0, 255),
+        (0, 255, 0),
+        (0, 0, 255),
+        (255, 255, 255),
+        (0, 0, 0),
+    ]
+
+    def __init__(
+        self,
+        root: str = "data",
+        split: str = "train",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new DeepGlobeLandCover dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train" or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+        """
+        assert split in self.splits
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self.checksum = checksum
+
+        self._verify()
+        if split == "train":
+            split_folder = "training_data"
+        else:
+            split_folder = "test_data"
+
+        self.files = []
+        for image in os.listdir(
+            os.path.join(root, self.data_root, split_folder, "images")
+        ):
+            if image.endswith(".jpg"):
+                id = image[:-8]
+                image_path = os.path.join(
+                    root, self.data_root, split_folder, "images", image
+                )
+                mask_path = os.path.join(
+                    root, self.data_root, split_folder, "masks", str(id) + "_mask.png"
+                )
```

Comment:
Since there is always a 1-to-1 mapping between image_path and mask_path, why not only store one in `self.files` (use a list instead of a dict) and convert the name on the fly? Then we store less data in memory and sorting becomes easier.

Reply:
I got rid of the dictionaries which I think is a small improvement. Doing string operations on the fly (i.e. every index into the dataset) will be a little slower than simple lookup and two lists of size "number of files" is tiny, so I don't think this is a problem.